Добавление примера магистерской диссертации

gorbunov-dmitry opened the pull request:
К примерам добавлен документ, отражающий базовую структуру магистерской диссертации. В документе помимо основного отчёта о работе содержатся:

- отзыв руководителя ВКР
- рецензия на ВКР
- задание на ВКР

gorbunov-dmitry committed:
Move practice example to example/practice

gorbunov-dmitry committed:
Add masters thesis example structure

gorbunov-dmitry committed:
Merge branch 'dvfu:master' into master

GoldFeniks reviewed (changes requested):
Презентация не является частью практики, поэтому должна быть перемещена в отдельную директорию

GoldFeniks commented on example/masters/README.md:
```diff
@@ -0,0 +1,10 @@
+# Пример магистерской диссертации
+
+Для сборки документа необходимо
```
Эти зависимости требуются только для сборки документов с листингами исходного кода с использованием пакета minted. Рекомендации по формлению кода стоит поместить где-то в другом месте или удалить вовсе

GoldFeniks commented on example/masters/README.md:
```diff
@@ -0,0 +1,10 @@
+* Python2 версии 2.6 и выше
+* Библиотека [Pygments](https://pypi.org/project/Pygments/)
+
+Рекомендуется использовать [TeXstudio](https://www.texstudio.org/) или команду 
```
Может быть информацию о сборке примеров разместить отдельно? Чтобы не повторять одно и то же

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\documentclass{../../cls/fefu}
+
+\usepackage{subfiles}
+\usepackage{amsmath}
+\usepackage{mathtools}
+\usepackage{amssymb}
+\usepackage{nicefrac}
+\usepackage{hyperref}
+\usepackage{minted}
+\usepackage{multirow}
+\usepackage{subcaption}
+\usepackage{placeins}
```
Ни один из этих пакетов не требуется примером, поэтому здесь их указывать не нужно

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\usepackage{multirow}
+\usepackage{subcaption}
+\usepackage{placeins}
+\addbibresource{references.bib}
+\graphicspath{{./images/}}
```
Данных файлов и папок не существует

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\addbibresource{references.bib}
+\graphicspath{{./images/}}
+
+\newcommand{\pa}[1]{\left(#1\right)}
+\newcommand{\code}[2][text]{\mintinline{#1}{#2}}
+\newcommand{\paramrow}[4][-]{\code{#2} & \code{#3} & \code{#1} & #4\\}
+\newcommand{\paramrowlistvalue}[4][]{\code{#2} & \code{#3} & \code{[#1]} & #4\\}
+\newcommand{\niceref}[2]{\hyperref[#1]{#2 \ref{#1}}}
+\DeclareMathOperator{\sech}{sech}
```
Ни один из этих макросов не используется примером, поэтому их не нужно определять

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\title{Тема работы}
+\setconsultant{Фамилия Имя Отчество}
+
+% \newif\ifreviewsonly
+\reviewsonlytrue
```
Это не имеет смысла в примере

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\disablehyphenation
+
+\begin{document}
+    \ifreviewsonly
```
Следует оставить только оформление дипломной работы. Отзывы удалить вовсе или поместить в отдельный документ (как презентацию)

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+\begin{document}
+    \ifreviewsonly
+        \begingroup
+            \setreviewer{Фамилия Имя Отчество}{должность, степень}
```
Это артефакт, здесь не требуется

GoldFeniks commented on example/masters/document.tex:
```diff
@@ -0,0 +1,74 @@
+        \endgroup
+    \else
+        \maketitle{thesis}
+        \makethesistitlebackside[1.2]
```
Парметр 1.2 не является обязательным, он нужен только в том случае, когда обратная сторона титульного листа не помещается на одну страницу

GoldFeniks commented on example/masters/subfiles/review.tex:
```diff
@@ -0,0 +1,15 @@
+\documentclass[../document.tex]{subfiles}
+
+\begin{document}
+    \begin{review}[Фамилия имя отчество]{}{01.01}
```
ФИО здесь в родительном падеже

GoldFeniks commented on example/masters/subfiles/supervisor_review.tex:
```diff
@@ -0,0 +1,7 @@
+\documentclass[../document.tex]{subfiles}
+
+\begin{document}
+    \begin{supervisorreview}[Фамилия Имя Отчество]{оценка}{баллы}{число}
```
ФИО здесь в родительном падеже

GoldFeniks commented on example/masters/subfiles/task.tex:
```diff
@@ -0,0 +1,20 @@
+\documentclass[../document.tex]{subfiles}
+
+\begin{document}
+    \begin{thesistask}[Фамилия Имя Отчество]{01.01.22}{Номер приказа}{01.01.22}{01.01.22}
```
ФИО здесь в родительном падеже

GoldFeniks commented on example/masters/subfiles/abstract.tex:
```diff
@@ -0,0 +1,8 @@
+
+\begin{document}
+	\begin{abstract}
+		\par ...
```
Многоточие здесь и во всех других местах кажутся избыточными

GoldFeniks commented on example/masters/subfiles/development.tex:
```diff
@@ -0,0 +1,7 @@
+
+\begin{document}
+    \section{Реализация}
+        \subsection{...}
```
Наличие заголовков второго уровня не является обязтаельным

GoldFeniks reviewed (changes requested):
Названия заголовков также следует уточнить по правилам оформления

gorbunov-dmitry committed:
Fix masters thesis example

- remove review and supervisor's review
- remove unused packages
- fix other minor mistakes
